consensus/types/src/execution_proof_subnet_id.rs

```diff
@@ -0,0 +1,138 @@
+//! Identifies each execution proof subnet by an integer identifier.
+use serde::{Deserialize, Serialize};
+use ssz::{Decode, DecodeError, Encode};
+use std::fmt::{self, Display};
+use std::ops::{Deref, DerefMut};
+
+/// Maximum number of execution proof subnets allowed by the protocol.
+///
+/// This is a hard protocol limit that defines the total number of proof subnets
+/// that can exist in the network. Individual nodes may choose to participate in
+/// fewer subnets (configured via max_execution_proof_subnets in ChainConfig),
+/// but no node can exceed this protocol maximum.
+///
+/// The value of 8 subnets provides a good balance between:
+/// - Proof diversity (multiple independent proofs per block)
+/// - Network overhead (not too many gossip topics)
+/// - Resource requirements (reasonable for most nodes(?))
+///
+/// In reality, I do not think we will have 8, more closer to 3, though this is still being
+/// explored. This number could be larger if we consider combining different zkVMs with different guests.
+pub const MAX_EXECUTION_PROOF_SUBNETS: u64 = 8;
+
+/// ExecutionProofSubnetId is both the id for the subnet that a particular proof will be on
+/// and the proof ID to identify the proof. ie, we have one type of proof per subnet.
+#[derive(arbitrary::Arbitrary, Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct ExecutionProofSubnetId(#[serde(with = "serde_utils::quoted_u64")] u64);
+
+impl ExecutionProofSubnetId {
+    /// Create an ExecutionProofSubnetId from a u64, validating it's within bounds
+    ///
+    /// Note: bounds here relates to the fact that there is a maximum number of subnets
+    /// that we can have; it is the number of maximum number of proofs that we will accept.
+    pub fn new(id: u64) -> Result<Self, InvalidSubnetId> {
```

Comment:
Main difference between this subnet wrapper type and others, like the ones for columns, is the fact that it has this validation check.